cmd/juju/application/deployer/charm.go

```diff
@@ -584,6 +592,40 @@ func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 	return c.deploy(ctx, deployAPI)
 }
 
+func (c *repositoryCharm) uploadExistingPendingResources(
+	appName string,
+	pendingResources []application.PendingResourceUpload,
+	api DeployerAPI) error {
```

Comment:
s/api DeployerAPI/apiCallCloser base.APICallCloser/

The entire DeployerAPI is not required here, just an APICallCloser, which is part of DeployerAPI